Skeleton based action recognition ros

negarhdr opened the pull request:
This pull request adds the ROS node of skeleton-based human action recognition (both baseline and progressive versions). 
The script is added in 
`projects/opendr_ws/src/perception/scripts/skeleton_based_action_recognition.py`

and a short description is added in: 
`projects/opendr_ws/src/perception/README.md`

and the list of nodes is updated in:
`projects/opendr_ws/README.md`

passalis commented on projects/opendr_ws/src/perception/scripts/skeleton_based_action_recognition.py:
```diff
@@ -0,0 +1,205 @@
+        # Run action recognition
+        category = self.action_classifier.infer(skeleton_seq)
+        category.confidence = category.confidence.max()
+        category.description = NTU60_ClASSES[category.data]
```
Shouldn't this be part of the original learner implementation?

negarhdr replied:
You are right. The inference implementation of learners is now modified and this part is removed from the ROS node implementation.

passalis commented on projects/opendr_ws/src/perception/scripts/skeleton_based_action_recognition.py:
```diff
@@ -0,0 +1,205 @@
+from opendr.perception.skeleton_based_action_recognition import SpatioTemporalGCNLearner
+from opendr.perception.skeleton_based_action_recognition import ProgressiveSpatioTemporalGCNLearner
+
+NTU60_ClASSES = pd.read_csv(
```
Does this work when the ros node is executed from the opendr_ws folder?

negarhdr replied:
This is removed from the implementation and added to the dataset part of the learners. So no need to load the dictionary anymore.

passalis reviewed (commented):
Thank you! Overall seems fine. I have two minor comments to be addressed. The first one is related to the way of loading the labels using a relative path (meaning that probably the node will not start correctly if not executed from the root). The second one is that the description is passed into the Category object outside the learning, which is not user-friendly (Category supports the description field to provide an easy way for the user to get the description without performing any kind of manual loading/translation)

negarhdr commented:
@ad-daniel the test is failing again because it is killed. We had a similar issue before because of a memory shortage.

ad-daniel commented:
Yes, but is very inconsistent, sometimes rerunning it works... We'll need a proper way of handling it, but for now let's see if this helps

passalis reviewed (commented):
Thank you for the changes!!! One last very minor comment: Should `['NTU60_ClASSES', 'KINETICS400_ClASSES']` be changed to `['NTU60_CLASSES', 'KINETICS400_CLASSES']`?

ad-daniel commented:
I assume it was a typo so I fixed it, but feel free to revert the commit if it wasn't (or just tell me and I will). I also simplified the import and the merge conflict. Hopefully the tests will pass on the first try, if so we can merge.

For now I'd leave running only 10 jobs in parallel, on some of the tests I made it appeared to help. At worst it just slows things down a bit, but re-running does it too so no loss there. When this issue https://github.com/tasostefas/opendr_internal/issues/167 is solved however we can remove this limitation.

negarhdr commented:
> I assume it was a typo so I fixed it, but feel free to revert the commit if it wasn't (or just tell me and I will). I also simplified the import and the merge conflict. Hopefully the tests will pass on the first try, if so we can merge.
> 
> For now I'd leave running only 10 jobs in parallel, on some of the tests I made it appeared to help. At worst it just slows things down a bit, but re-running does it too so no loss there. When this issue #167 is solved however we can remove this limitation.

Yes it was a typo and thank you for fixing it. The tests are passed as well.

passalis reviewed (approved):
Thank you!

ad-daniel reviewed (dismissed):
Thank you!

negarhdr commented:
@ad-daniel , @passalis 
a small update is just pushed to this branch. I need your approval again to merge.
Thanks.
